build-tools/scripts/MSBuildReferences.projitems

```diff
@@ -20,4 +20,8 @@
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
     </None>
   </ItemGroup>
+  <!-- Copy PackageReference content (as specified above) to OutputDir. This no longer happens by default in short-form projects. -->
+  <PropertyGroup>
+      <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
+  </PropertyGroup>
 </Project>
```

Comment:
@pjcollins So I looked at the issue with the tests not running in VSForMac and I had to apply this patch https://gist.github.com/dellis1972/e9961b958e78c3ee45ed608617cb1335. Not sure if this is something you want to include in the PR or if I should do a separate one. 

Oddly the tests work fine in VSCode on Mac without this change.

Reply:
I'd rather keep that patch in a separate PR. As things stand now the scope of change is probably already too large, and I am considering splitting up the changes in this PR further.